test/csuite/wt12015_backup_corruption/main.c

```diff
@@ -0,0 +1,473 @@
+    testutil_assert_errno((pid = fork()) >= 0);
+
+    if (pid == 0) { /* Child. */
+        testutil_recreate_dir(WT_HOME_DIR);
+        testutil_wiredtiger_open(
+          opts, WT_HOME_DIR, env_config, &child_event_handler, &conn, false, false);
+        testutil_check(conn->open_session(conn, NULL, NULL, &session));
+        testutil_check(session->create(session, TABLE_URI, TABLE_CONFIG));
+
+        /* Do some work, while creating checkpoints and doing backups. */
+        do_work_before_failure(conn, session);
+
+        /* Die before finishing the next checkpoint. */
+        printf("Setting the failpoint...\n");
+        testutil_check(
+          session->reconfigure(session, "debug=(checkpoint_fail_before_turtle_update=true)"));
+        testutil_sentinel(NULL, EXPECT_ABORT);
+        testutil_check(session->checkpoint(session, NULL));
+        testutil_remove(EXPECT_ABORT);
+
+        /* We should die before we get here. */
+        testutil_die(ENOTRECOVERABLE, "The child process was supposed be dead by now!");
+    }
+
+    /* Parent. */
+
+    /* Wait for the child to die. */
+    testutil_assert(waitpid(pid, &status, 0) > 0);
+    printf("-- crash --\n");
+
+    /* Save the database directory. */
+    testutil_copy(WT_HOME_DIR, "save");
+
+    /* Do more work, create another backup, and verify it. */
+    do_work_after_failure(true);
+}
+
+/*
+ * run_test_force_stop --
+ *     Run a test with force stop.
+ */
+static void
+run_test_force_stop(void)
+{
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+    pid_t pid;
+    int status;
+
+    printf("\n%s: Test crashing during force-stop after incremental backup\n", __func__);
+
+    testutil_remove(EXPECT_ABORT);
+    testutil_assert_errno((pid = fork()) >= 0);
+
+    if (pid == 0) { /* Child. */
+        testutil_recreate_dir(WT_HOME_DIR);
+        testutil_wiredtiger_open(
+          opts, WT_HOME_DIR, env_config, &child_event_handler, &conn, false, false);
+        testutil_check(conn->open_session(conn, NULL, NULL, &session));
+        testutil_check(session->create(session, TABLE_URI, TABLE_CONFIG));
+
+        /* Do some work, while creating checkpoints and doing backups. */
+        do_work_before_failure(conn, session);
+
+        /* Die before finishing the next checkpoint. */
+        printf("Setting the failpoint...\n");
+        testutil_check(
+          session->reconfigure(session, "debug=(checkpoint_fail_before_turtle_update=true)"));
+        testutil_sentinel(NULL, EXPECT_ABORT);
+        testutil_backup_force_stop(session);
+        testutil_remove(EXPECT_ABORT);
+
+        /* We should die before we get here. */
+        testutil_die(ENOTRECOVERABLE, "The child process was supposed be dead by now!");
+    }
+
+    /* Parent. */
+
+    /* Wait for the child to die. */
+    testutil_assert(waitpid(pid, &status, 0) > 0);
+    printf("-- crash --\n");
+
+    /* Save the database directory. */
+    testutil_copy(WT_HOME_DIR, "save");
```

Comment:
One more minor question: If we run with both tests, this one will find `save` already exists with older files from `run_test_backup`. (It isn't clear from this code if `save` is under HOME or at the same level as HOME. If we copy now over the other copy, `save` is not a valid database. Some possible suggestions:
1. Name them `save_backup` and `save_force_stop`.
2. Or remove `save` if it exists before doing the copy.
3. If this cannot happen because `save` is under HOME then maybe just change the comment to "Save the database directory to HOME/save."

Reply:
Good catch! Yes, this would in fact reuse the same "save" directory between the two tests, which is created under the test directory (starting with "WT_TEST."), but it will be nonetheless shared between the two tests. I opted for the first suggestion, which is to use different names.